Lib/test/test_exceptions.py

```diff
@@ -1014,6 +1014,9 @@ def cycle():
 
     def test_no_hang_on_context_chain_cycle2(self):
         # See issue 25782. Cycle at head of context chain.
```

Comment:
This fix looks incorrect, tests should not depend on the GC to pass. When this happens, is a symptom of another problem. 

I propose to revert this commit and investigate.

Reply:
CC: @vstinner

Reply:
I understand is a "temporary measure" but in my experience those are left there with no fixes more often than not. Also, I don't feel comfortable with "
temporary fixes in the release candidate.

Reply:
The alternative is to disable the test. That doesn't fix the issue either.

Reply:
I prefer to deactivate the test. The reason is that relying on the GC in this way at the end has global effects and can mask other issues. Is also not deterministic and can actually be an endless loop in some extreme situations involving resurrection.

This is just my opinion on this of course, If the consensus is to leave this because the test has more value, then let's leave it, but I have to say that my previous experience with these kind of fixes is that they are left there more often than not.

Reply:
Sure. I wasn't sure myself which way to go. I agree that disabling the test is clearer.

I'm out now so will do this later today when I'm back. Unless someone beats me to it.

Reply:
Don't worry, is not urgent.

Thanks a lot for the investigation and for all the work!! 🚀

Reply:
> I don't feel comfortable with " temporary fixes in the release candidate.

Sure, we weren't going to let this slip into RC2. The point was to make refleak tests able to catch other regressions on that branch in the mean time.

Reply:
If you'd rather redo the fix as a skip instead of `gc.collect()` then that's fine as well. However, from what I understood on the PR, having it run on the entire buildbot fleet for a few days would give us more confidence whether that approach to working around the refleaks is even effective.

How about we leave it as is for the weekend and remove the `gc.collect()` loop on Monday?

Reply:
> If you'd rather redo the fix as a skip instead of `gc.collect()` then that's fine as well. However, from what I understood on the PR, having it run on the entire buildbot fleet for a few days would give us more confidence whether that approach to working around the refleaks is even effective.
> 


I don't get what you mean by this. Why do we want to know if the approach to work around is effective? What information do we gain by this?   I can understand the though that this may gives us some more light into the problem but this workaround is too intrusive to gather any conclusions from the actual problem, more then that a cycle is likely involved.

> How about we leave it as is for the weekend and remove the `gc.collect()` loop on Monday?

👍 Works for me

Reply:
> Why do we want to know if the approach to work around is effective?

Irit [wrote](https://github.com/python/cpython/pull/27746#issuecomment-897807162):

> Let's run this on buildbots and if it indeed resolves the leak we can do this for now.

I misinterpreted this as "let's merge this and see" but obviously she meant the `test-with-buildbots` label. Nevermind!